test/functional/wallet_conflicts.py

```diff
@@ -123,5 +129,148 @@ def run_test(self):
+
+        assert_equal(alice.gettransaction(tx1)["mempoolconflicts"], [tx2])
+
+        self.log.info("Test scenario where a mempool conflict is removed")
+
+        tx3 = alice.sendrawtransaction(tx3)
+
+        assert_equal(alice.gettransaction(tx1)["mempoolconflicts"], [])
+
+        # now all of alice's outputs should be considered spent
+        assert_equal(alice.listunspent(), [])
+        assert_equal(alice.getbalance(), 0)
+
+        alice.sendrawtransaction(alice.gettransaction(tx1)['hex'])
+        self.generate(self.nodes[0], 3)
+        assert_equal(alice.getbalance(), 75)
+
+    def test_mempool_and_block_conflicts(self):
+        alice = self.nodes[0].get_wallet_rpc("alice")
+        bob = self.nodes[1].get_wallet_rpc("bob")
+
+        self.log.info("Test a scenario where a transaction has both a block conflict and a mempool conflict")
+        unspents = [{"txid" : element["txid"], "vout" : element["vout"]} for element in alice.listunspent()]
+
+        assert_equal(bob.getbalances()["mine"]["untrusted_pending"], 0)
+
+        self.disconnect_nodes(0, 1)
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[0]], outputs=[{bob.getnewaddress() : 24.99999}])
+        raw_tx1 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+        tx1 = bob.sendrawtransaction(raw_tx1)
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[0], unspents[2]], outputs=[{alice.getnewaddress() : 49.999}])
+        tx1_conflict = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[2]], outputs=[{alice.getnewaddress() : 24.99}])
+        tx1_conflict_conflict = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[1]], outputs=[{bob.getnewaddress() : 24.9999}])
+        raw_tx2 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+        tx2 = bob.sendrawtransaction(raw_tx2)
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[1]], outputs=[{alice.getnewaddress() : 24.9999}])
+        tx2_conflict = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        bob_unspents = [{"txid" : element, "vout" : 0} for element in [tx1, tx2]]
+
+        assert_equal(bob.getbalances()["mine"]["untrusted_pending"], Decimal("49.99989000"))
+
+        raw_tx = bob.createrawtransaction(inputs=[bob_unspents[0], bob_unspents[1]], outputs=[{bob.getnewaddress() : 49.999}])
+        raw_tx3 = bob.signrawtransactionwithwallet(raw_tx)['hex']
+        tx3 = bob.sendrawtransaction(raw_tx3)
+
+        # alice has 0 txs, bob has 3
+        assert_equal(len(alice.getrawmempool()), 0)
+        assert_equal(len(bob.getrawmempool()), 3)
+
+        assert_equal(bob.getbalances()["mine"]["untrusted_pending"], Decimal("49.99900000"))
+
+        # bob broadcasts tx_1 conflict
+        tx1_conflict = bob.sendrawtransaction(tx1_conflict)
+        assert_equal(len(alice.getrawmempool()), 0)
+        assert_equal(len(bob.getrawmempool()), 2)
```

Comment:
In 003efbbe45079c4416810a025b2bc372559dff15 "test: Add tests for wallet mempool conflicts":
`tx1_conflict` should displace `tx1`, but why does Bob not have 3 again now? Shouldn’t Bob know about `tx1`, `tx2`, and `tx3` before, and then `tx1_conflict`, `tx2`, and `tx3` after?

Reply:
`tx1_conflict` also displaces `tx3` because `tx3` spends `tx1`, so it is also considered mempool-conflicted. So instead `tx1_conflict` and `tx2` are the only transactions left in the mempool. I've added a comment indicating this.